File: briar-android/src/test/java/org/briarproject/briar/android/telegram/TelegramConversationMapperTest.java
package org.briarproject.briar.android.telegram;

import org.briarproject.briar.api.telegram.TelegramMessage;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class TelegramConversationMapperTest {

	@Test
	public void testTextMessagesRenderInDateOrderOnly() {
		List<TelegramConversationUiMessage> items =
				TelegramConversationMapper.toUiMessages(Arrays.asList(
						new TelegramMessage(1L, 3L, 30, true, "new"),
						new TelegramMessage(1L, 2L, 20, false, ""),
						new TelegramMessage(1L, 4L, 40, false, " \t\n"),
						new TelegramMessage(1L, 1L, 10, false, "old")
				));

		assertEquals(2, items.size());
		assertEquals("old", items.get(0).getText());
		assertEquals(10000L, items.get(0).getDateMillis());
		assertFalse(items.get(0).isOutgoing());
		assertEquals("new", items.get(1).getText());
		assertEquals(30000L, items.get(1).getDateMillis());
		assertTrue(items.get(1).isOutgoing());
	}
}

File: briar-android/src/test/kotlin/org/briarproject/briar/android/telegram/TelegramConversationMapperTest.kt
package org.briarproject.briar.android.telegram

import org.briarproject.briar.api.telegram.TelegramMessage
import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class TelegramConversationMapperTest {

	@Test
	fun testTextMessagesRenderInDateOrderOnly() {
		val items = TelegramConversationMapper.toUiMessages(listOf(
			TelegramMessage(1L, 30L, 30, true, "new").toConnectorMessage(),
			TelegramMessage(1L, 2L, 20, false, "").toConnectorMessage(),
			TelegramMessage(1L, 4L, 40, false, " \t\n").toConnectorMessage(),
			TelegramMessage(1L, 1L, 10, false, "old").toConnectorMessage(),
			TelegramMessage(1L, 20L, 30, false, "same time").toConnectorMessage(),
		))

		assertEquals(3, items.size)
		assertEquals("telegram:1:1", items[0].stableId)
		assertEquals("telegram", items[0].connectorSource.id)
		assertEquals("1", items[0].connectorThreadId)
		assertEquals("1", items[0].connectorMessageId)
		assertEquals("old", items[0].text)
		assertEquals(10000L, items[0].dateMillis)
		assertFalse(items[0].isOutgoing)
		assertEquals("telegram:1:20", items[1].stableId)
		assertEquals("same time", items[1].text)
		assertEquals(30000L, items[1].dateMillis)
		assertFalse(items[1].isOutgoing)
		assertEquals("telegram:1:30", items[2].stableId)
		assertEquals("new", items[2].text)
		assertEquals(30000L, items[2].dateMillis)
		assertTrue(items[2].isOutgoing)
	}
}
